maddymz/irondb
# US870

US adds approve page so user can approve a pending entry.
**Note:** I tried using the mock data to make this easier to test, but ran into issues due to the user constraint. 
**Note 2:** I may have uncovered a bug where the authors add button isn't working on the editor, if this is the case it's in dev as I have not altered the file here. I will address that when refactoring the editor/data-entry in my next user story.

To test: 

1. `git checkout US870`
2. Be sure you're using new db-init so run `
3. Run `./iron.sh -pm`
4. Navigate to `localhost:3001/login`
5. Login as `user1` with password `password`
6. Notice panel looks like: 
![Screen Shot 2019-04-03 at 6 29 51 PM](https://user-images.githubusercontent.com/6512755/55518745-27ada600-5643-11e9-92dd-b4d4b5411e6c.jpg)

7. Go to data entry and select manual without pdf, input an entry:
![Screen Shot 2019-04-03 at 6 32 11 PM](https://user-images.githubusercontent.com/6512755/55518768-372cef00-5643-11e9-9d18-b5c14f6db16b.jpg)

8. Notice updated panel table:
![Screen Shot 2019-04-03 at 6 32 27 PM](https://user-images.githubusercontent.com/6512755/55518787-4c098280-5643-11e9-8c30-310d831bb3b8.jpg)

9. Go to data entry and select manual with pdf, input an entry (pdf not loading in firefox still an issue tbd):
![Screen Shot 2019-04-03 at 6 35 50 PM](https://user-images.githubusercontent.com/6512755/55518816-6a6f7e00-5643-11e9-9820-8074ea514db2.jpg)

10. Again notice updated panel:
![Screen Shot 2019-04-03 at 6 36 13 PM](https://user-images.githubusercontent.com/6512755/55518831-79563080-5643-11e9-9057-9d9e6bfaa59c.jpg)

11. Click on link to unapproved page from panel:
![Screen Shot 2019-04-03 at 6 36 22 PM](https://user-images.githubusercontent.com/6512755/55518847-8d019700-5643-11e9-9c40-2faef7f34007.jpg)

12. Select load approval, should be blocked as these are your own entries:
![Screen Shot 2019-04-03 at 6 40 13 PM](https://user-images.githubusercontent.com/6512755/55518861-a0acfd80-5643-11e9-8e51-4c5cf3b9b527.jpg)

13. Logout, log back in with `user2` and password `password`
14. Notice panel updated:
![Screen Shot 2019-04-03 at 6 40 50 PM](https://user-images.githubusercontent.com/6512755/55518905-bd493580-5643-11e9-979c-e67d688da47c.jpg)

15. Select approval needed from panel and load the entry without pdf for approval:
![Screen Shot 2019-04-03 at 6 41 02 PM](https://user-images.githubusercontent.com/6512755/55518949-e5389900-5643-11e9-9b11-60deb1eb5342.jpg)

16. Approval all parts, expected message:
![Screen Shot 2019-04-03 at 6 41 14 PM](https://user-images.githubusercontent.com/6512755/55518970-faadc300-5643-11e9-985e-e4b4a669e9cc.jpg)

17. Load for approval entry with pdf:
![Screen Shot 2019-04-03 at 6 41 33 PM](https://user-images.githubusercontent.com/6512755/55518982-09947580-5644-11e9-8b75-ccdf477aa3e3.jpg)

18. Notice pdf is fixed as per new pdf usage:
![Screen Shot 2019-04-03 at 6 41 43 PM](https://user-images.githubusercontent.com/6512755/55518995-1b761880-5644-11e9-9b63-9f93b96c73c8.jpg)

19. Example of expected success feedback for each section:
![Screen Shot 2019-04-03 at 6 41 59 PM](https://user-images.githubusercontent.com/6512755/55518999-24ff8080-5644-11e9-8e52-26ff4b127863.jpg)

20. Complete approval:
![Screen Shot 2019-04-03 at 6 42 12 PM](https://user-images.githubusercontent.com/6512755/55519042-47919980-5644-11e9-8d1c-80b5f4d45db4.jpg)

21. Select needs approval, expected message for no pending entries:
![Screen Shot 2019-04-03 at 6 52 01 PM](https://user-images.githubusercontent.com/6512755/55519075-609a4a80-5644-11e9-8ef9-5a8edade07fd.jpg)

22. Go to database page to see live entries, find the entries you just approved:
![Screen Shot 2019-04-03 at 7 16 40 PM](https://user-images.githubusercontent.com/6512755/55519279-25e4e200-5645-11e9-9597-934eaaede83d.jpg)

Success!!!
